joomla/joomla-cms
# [4.0] Smart Search categories plugin needs to use right extension element

This is a redo of PR #16278.

When using the advanced features of the category system, specifically several types of categories per component, the Smart Search categories plugin fails and uses the wrong parts of the extension. This should fix that.

### Testing Instructions
1. Make sure that the `Content - Smart Search` plugin is enabled.
2. Go to `administrator/index.php?option=com_categories&view=categories&extension=com_content.test` of your 4.0 test installation. This simulates that you have a category type `test` for a component.
3. Click on "New" and create a category with a unique name, so that you can find it again in the Smart Search backend. Alternatively you can clear the index of Smart Search first.
4. Save the new category and then go to the Smart Search index. See that the category has NOT been indexed.
5. Apply the patch, go to the above URL again, edit the category that we just created and save it again.
6. Go to the Smart Search again and see that the category has now been indexed.
